owncloudApp/src/main/java/com/owncloud/android/presentation/ui/toolbar/ToolbarActivity.kt

```diff
@@ -0,0 +1,120 @@
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.presentation.ui.toolbar
+
+import android.content.Intent
+import android.view.View
+import android.view.View.VISIBLE
+import android.widget.TextView
+import androidx.appcompat.widget.SearchView
+import androidx.core.view.isVisible
+import com.google.android.material.imageview.ShapeableImageView
+import com.google.android.material.textview.MaterialTextView
+import com.owncloud.android.R
+import com.owncloud.android.authentication.AccountUtils
+import com.owncloud.android.ui.activity.BaseActivity
+import com.owncloud.android.ui.activity.ManageAccountsActivity
+import com.owncloud.android.utils.AvatarUtils
+import kotlinx.android.synthetic.main.owncloud_toolbar.*
+
+/**
+ * Base class providing toolbar registration functionality, see [.setupToolbar].
+ */
+abstract class ToolbarActivity : BaseActivity() {
+
+    /**
+     * Toolbar setup that must be called in implementer's [.onCreate] after [.setContentView] if they
+     * want to use the toolbar.
+     */
+    open fun setupToolbar(
+        toolbarConfig: ToolbarConfig
+    ) {
+        when (toolbarConfig) {
+            is ToolbarConfig.ToolbarStandard -> {
+                configStandardToolbar(toolbarConfig)
+            }
+            is ToolbarConfig.ToolbarRoot -> {
+                configRootToolbar(toolbarConfig)
+            }
+        }
+    }
+
+    private fun configStandardToolbar(standardToolbar: ToolbarConfig.ToolbarStandard) {
+        useStandardToolbar(true)
+
+        standardToolbar.title?.let { standard_toolbar?.title = it }
+        setSupportActionBar(standard_toolbar)
+        supportActionBar?.setDisplayHomeAsUpEnabled(standardToolbar.displayHomeAsUpEnabled)
+        supportActionBar?.setHomeButtonEnabled(standardToolbar.homeButtonEnabled)
+        supportActionBar?.setDisplayShowTitleEnabled(standardToolbar.title != null)
+    }
+
+    private fun configRootToolbar(rootToolbar: ToolbarConfig.ToolbarRoot) {
+        useStandardToolbar(false)
+
+        val toolbarTitle = findViewById<TextView>(R.id.root_toolbar_title)
+        val searchView = findViewById<SearchView>(R.id.root_toolbar_search_view)
+        val avatarView = findViewById<ShapeableImageView>(R.id.root_toolbar_avatar)
+
+        with(toolbarTitle) {
+            isVisible = true
+            text = rootToolbar.title
+            if (rootToolbar.enableSearch) {
+                setOnClickListener {
+                    toolbarTitle.isVisible = false
+                    searchView.isVisible = true
+                    searchView.isIconified = false
+                }
+            }
+        }
+
+        with(searchView) {
+            isVisible = false
+            setOnCloseListener {
+                searchView.visibility = View.GONE
+                toolbarTitle.visibility = VISIBLE
+                false
+            }
+        }
+
+        with(avatarView) {
```

Comment:
Yes, it is needed to set the onClickListener.

Reply:
I mean this:

![Screenshot_2021-02-16_at_11_08_32](https://user-images.githubusercontent.com/15261363/108048855-d4cb1500-7047-11eb-82d5-596aaba972f3.png)

You are using apply but later writing `avatarView` again instead of `this`

Besides, `AccountUtils.getCurrentOwnCloudAccount(context) ?: return` doesn't use `avatarView` so I would call it out of the `apply`.

I would not add `AvatarUtils().loadAvatarForAccount(` call to the `apply` either, because we are not operating on the members of the receiver object, i.e. configuring the `avatarView` with that call.

```
AccountUtils.getCurrentOwnCloudAccount(context) ?: return

with(avatarView) {
   AvatarUtils().loadAvatarForAccount(
        avatarView,
        AccountUtils.getCurrentOwnCloudAccount(context),
        true,
        context.resources.getDimension(R.dimen.toolbar_avatar_radius)
    )
}

avatarView.setOnClickListener {
        startActivity(Intent(context, ManageAccountsActivity::class.java))
    }
}
```

Do you see the difference?